tests/standalone_tests/test_celery_app.py

```diff
@@ -0,0 +1,389 @@
+    )
+
+    assert (
+        initial_cel_app
+        == CeleryAppFactory().get_celery_app(RABBITMQ_GLOBALNODE_ADDR)._celery_app
+    ), "Celery app is different after the task was queued and fetched."
+
+
+@pytest.mark.slow
+@pytest.mark.integration
+def test_celery_app_get_task_timeout(
+    globalnode_node_service,
+    reset_celery_app_factory,
+    controller_testing_logger,
+):
+    cel_app_wrapper = CeleryAppFactory().get_celery_app(RABBITMQ_GLOBALNODE_ADDR)
+    initial_cel_app = cel_app_wrapper._celery_app
+
+    async_res = queue_slow_udf(cel_app_wrapper, controller_testing_logger)
+
+    with pytest.raises(CeleryTaskTimeoutException):
+        cel_app_wrapper.get_result(
+            async_result=async_res, timeout=1, logger=controller_testing_logger
+        )
+
+    assert (
+        initial_cel_app
+        == CeleryAppFactory().get_celery_app(RABBITMQ_GLOBALNODE_ADDR)._celery_app
+    ), "Celery app is different after the task timed out, even though the node never went down."
+
+
+@pytest.mark.integration
+def test_celery_app_queue_task_with_rabbitmq_down(
+    reset_celery_app_factory,
+    controller_testing_logger,
+):
+    cel_app_wrapper = CeleryAppFactory().get_celery_app(RABBITMQ_LOCALNODETMP_ADDR)
+    initial_cel_app = cel_app_wrapper._celery_app
+
+    with pytest.raises(CeleryConnectionError):
+        cel_app_wrapper.queue_task(
+            task_signature=get_celery_task_signature("get_node_info"),
+            logger=controller_testing_logger,
+            request_id=request_id,
+        )
+
+    assert (
+        initial_cel_app
+        == CeleryAppFactory().get_celery_app(RABBITMQ_LOCALNODETMP_ADDR)._celery_app
+    ), "The celery apps should be the same, the rabbitmq is still unreachable."
+
+
+@pytest.mark.integration
+def test_celery_app_get_task_res_with_rabbitmq_down(
+    reset_celery_app_factory,
+    controller_testing_logger,
+):
+    cel_app_wrapper = CeleryAppFactory().get_celery_app(RABBITMQ_LOCALNODETMP_ADDR)
+    initial_cel_app = cel_app_wrapper._celery_app
+
+    with pytest.raises(CeleryConnectionError):
+        cel_app_wrapper.get_result(
+            async_result=AsyncResult(app=cel_app_wrapper._celery_app, id="whatever"),
+            timeout=10,
+            logger=controller_testing_logger,
+        )
+
+    assert (
+        initial_cel_app
+        != CeleryAppFactory().get_celery_app(RABBITMQ_LOCALNODETMP_ADDR)._celery_app
+    ), "The celery app should reset, there was a connectivity error."
+
+
+@pytest.mark.slow
+@pytest.mark.integration
+def test_celery_app_get_task_res_with_node_down(
+    localnodetmp_node_service,
+    reset_celery_app_factory,
+    controller_testing_logger,
+):
+    cel_app_wrapper = CeleryAppFactory().get_celery_app(RABBITMQ_LOCALNODETMP_ADDR)
+    initial_cel_app = cel_app_wrapper._celery_app
+
+    async_res = queue_slow_udf(cel_app_wrapper, controller_testing_logger)
+
+    kill_service(localnodetmp_node_service)
```

Comment:
The reason this is different is that `queuing` a task doesn't have a problem if the rabbitmq restarted. It will work. So there is no reason to "initialize" the celery app.

On the other hand, `get_result` needs to have the celery app initialized, by being used at least once to get a result, in order for it to fail afterwards.

I should probably add documentation about these. Do you think it should be here in the tests or in the celery_app?

Reply:
yes try to add some comments in celery_app.py

Reply:
I saw now that I have this message in the celery_app
```
            # The celery app needs to be recreated due to a bug:
            # https://github.com/celery/celery/issues/6912
            # Create a new celery app since the current one is corrupted.
```

The ticket pretty much describes our problem.

I have changed it to a bit more explanatory:
```
            # The celery app needs to be recreated due to a bug:
            # https://github.com/celery/celery/issues/6912
            # If we don't reset the celery app queuing a task will work, but then we won't be able to fetch its result.
```